### PR TITLE
Added CustomJSONFormatter to generate Cucumber Reports for executed test cases in an event of Thread Failure

### PR DESCRIPTION
**Pull request by Subrahmanyam83**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,4 @@ release.properties
 dependency-reduced-pom.xml
 buildNumber.properties
 .mvn/timing.properties
+*.iml
```
Comment (sponte) [inline .gitignore]: It's ok for .iml to be in the repo

File: src/main/java/eu/evops/maven/pluins/cucumber/parallel/CustomJSONFormatter.java [hunk 2]
```diff
@@ -0,0 +1,250 @@
+package eu.evops.maven.pluins.cucumber.parallel;
```
Comment (sponte) [inline src/main/java/eu/evops/maven/pluins/cucumber/parallel/CustomJSONFormatter.java]: Is there a reason why this class does not inherit from JSONFormatter class and only override what it needs?

File: src/main/java/eu/evops/maven/pluins/cucumber/parallel/CustomJSONFormatter.java [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by n471306 on 06/07/2017.
+ */
+public class CustomJSONFormatter implements Reporter, Formatter {
```
Comment (sponte) [inline src/main/java/eu/evops/maven/pluins/cucumber/parallel/CustomJSONFormatter.java]: CustomerJSONFormatter name is not descriptive, can you rename to something more meaningful for instance StreamingJSONFormatter and adding appropriate comments to indicate why it's been added?

File: src/main/java/eu/evops/maven/pluins/cucumber/parallel/Run.java [hunk 4]
```diff
@@ -219,7 +219,10 @@ public void execute() throws MojoExecutionException, MojoFailureException {
     private void combineReports() throws MergeException, MojoFailureException {
         for (String plugin : plugins) {
             String pluginName = plugin.split(":")[0];
-            if(pluginName.matches("^(json|junit)")) {
+            if(pluginName.matches("^(CustomJSONFormatter|junit)")) {
```
Comment (sponte) [inline src/main/java/eu/evops/maven/pluins/cucumber/parallel/Run.java]: This code in unnecessary. You should update Merger.get method to return correct merge plugin

File: src/main/java/eu/evops/maven/pluins/cucumber/parallel/Run.java [hunk 5]
```diff
@@ -294,11 +297,18 @@ private void generateReportForJsonFiles(File reportOutputDirectory,
             args.add(CucumberArguments.Plugin.getArg());
 
             // If plugin ends with semicolon, I will generate report name under thread folder
-            if(plugin.endsWith(":")) {
-                File threadedReportFile = new File(threadFolder, "reports/" +
-                        getReportFileName(plugin.split(":")[0]));
-                args.add(format("%s%s", plugin, threadedReportFile.getAbsolutePath()));
-            } else {
+            if(plugin.endsWith(":")&&!plugin.startsWith("json")) {
```
Comment (sponte) [inline src/main/java/eu/evops/maven/pluins/cucumber/parallel/Run.java]: Again, this should not be necessary, update getReportFileName method that expects your new json reporter plugin name (absolute)

**Comment by Subrahmanyam83** [comment]
Please provide review comments

**Review by sponte — CHANGES_REQUESTED** [review]

File: src/main/java/eu/evops/maven/pluins/cucumber/parallel/Run.java [hunk 6]
```diff
@@ -361,6 +373,13 @@ private void setThreadCount() {
                 String.valueOf(threadCount));
     }
 
+    /* Returns the streamingFormatter value.
+    * if true, it uses the StreamingJSON Formatter
+    * else it uses the default Cucumber JSON Formatter*/
+    private boolean getStreamingFormatterValue() throws IOException {
```
Comment (sponte) [inline src/main/java/eu/evops/maven/pluins/cucumber/parallel/Run.java]: Why does this throw IOException?

File: pom.xml [hunk 7]
```diff
@@ -1,213 +1,392 @@
...
+				</groupId>
+				<artifactId>
+					maven-compiler-plugin
+				</artifactId>
+				<version>
+					3.3
+				</version>
+				<configuration>
+					<source>
+						1.8
```
Comment (sponte) [inline pom.xml]: Hey, this should revert back to 1.7 please

**Review by sponte — APPROVED** [review]

**Pull request merged**